component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1533,15 +1533,24 @@ private void handleSmsOtpVerificationFail(AuthenticationContext context) throws
         Map<String, String> updatedClaims = new HashMap<>();
         if ((currentAttempts + 1) >= maxAttempts) {
             // Calculate the incremental unlock-time-interval in milli seconds.
-            unlockTimePropertyValue = (long) (unlockTimePropertyValue * 1000 * 60 * Math.pow(unlockTimeRatio,
-                    failedLoginLockoutCountValue));
+            if (context.getProperty(SMSOTPConstants.OVERALL_FAILED_LOGIN_LOCKOUT_COUNT) != null) {
+                int overallLockoutCount =
+                        (int) context.getProperty(SMSOTPConstants.OVERALL_FAILED_LOGIN_LOCKOUT_COUNT);
+                unlockTimePropertyValue = (long) (unlockTimePropertyValue * 1000 * 60 * Math.pow(unlockTimeRatio,
+                        overallLockoutCount));
+                updatedClaims.put(SMSOTPConstants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM,
+                        String.valueOf(overallLockoutCount + 1));
+            } else {
+                unlockTimePropertyValue = (long) (unlockTimePropertyValue * 1000 * 60 * Math.pow(unlockTimeRatio,
+                        failedLoginLockoutCountValue));
+                updatedClaims.put(SMSOTPConstants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM,
+                        String.valueOf(failedLoginLockoutCountValue + 1));
+            }
```

Comment:
Let's use NumberUtils and Integer/Long classes to convert these strings to numbers.

Reply:
Resolved in https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/153/commits/9d2145ddc413e526a97f931bf0da0efb440269c2